tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -313,12 +314,20 @@ int main(int argc, char **argv)
      */
     {
         DEFER_CLEANUP(struct s2n_cert_chain_and_key *cert = NULL, s2n_cert_chain_and_key_ptr_free);
-        /* use a very insecure cert that would not be included in any reasonable cert preferences */
-        EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&cert, "rsae", "pkcs", "1024", "sha1"));
+        if (s2n_is_in_fips_mode() && !s2n_libcrypto_is_awslc()) {
+            /* OpenSSL FIPS module requires the key size for RSA to be bigger than or equal to 1024 bits.
+             * See https://github.com/aws/s2n-tls/issues/4651 for more information.
+             */
+            EXPECT_FAILURE_WITH_ERRNO(s2n_test_cert_permutation_load_server_chain(&cert, "rsae", "pkcs", "512", "sha1"),
+                    S2N_ERR_INVALID_RSA_KEY_SIZE_IN_OSSL_FIPS_MODE);
+        } else {
+            /* use a very insecure cert that would not be included in any reasonable cert preferences */
```

Comment:
This comment would be for both branches of the if/else, not just this one. We'd only be using 512 because it's even more insecure.